libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -0,0 +1,881 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author julius <djudju@protonmail.com>
+ * @date 2019
+ *  Converts the AST from JSON format to ASTNode
+ */
+
+#include <libsolidity/ast/ASTJsonImporter.h>
+#include <liblangutil/Scanner.h>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+#include <liblangutil/Token.h>
+#include <libyul/AsmParser.h>
+#include <libyul/backends/evm/EVMDialect.h>
+#include <liblangutil/SourceLocation.h>
+#include <liblangutil/Exceptions.h>
+#include <liblangutil/ErrorReporter.h>
+
+
+using namespace std;
+
+using namespace dev;
+using namespace solidity;
+
+using SourceLocation = langutil::SourceLocation;
+
+template<class T>
+ASTPointer<T> castPointer(ASTPointer<ASTNode> _ast)
```

Comment:
`const&`?

Reply:
also maybe: pointerCastChecked?

Reply:
Actually it might make more sense to create an template "overload" of convertJsonToAstNode that takes the actual ast node type as a parameter and performs the type conversion.

Reply:
> Actually it might make more sense to create an template "overload" of convertJsonToAstNode that takes the actual ast node type as a parameter and performs the type conversion.

Ah yes, we never use castPointer without invoking convertJsonToAstNode as its argument....might as well combine them. 

Done